### ISSUE TITLE
🐛 [firebase_messaging] flutter build ios hangs when building with code signing enabled

### ISSUE THREAD

**Issue by daniil-shumko** [FirebaseExtended/flutterfire#5121]
## Bug report

**Describe the bug**
NOTE: This bug only happens when running the `flutter build ios` command on CI, in my example that is GitHub Actions. 

`flutter build ios` command hangs when building the app on CI after adding `firebase_messaging: ^7.0.3` dependency.
If `--no-codesign` flag is set, the app builds fine on CI.

I have created a quick sample app that can be [viewed here](https://github.com/daniil-shumko/flutter_ios_build_issue).
Also, [GitHub Actions is configured](https://github.com/daniil-shumko/flutter_ios_build_issue/actions/workflows/flutter-ci.yml) for this repo and you can see all the results from different configurations with all the log outputs.

### Steps to reproduce

1. Create an empty flutter app with `flutter create` command
2. Add `firebase_messaging: ^7.0.3` to `pubspec.yaml` file
3. Configure iOS app signing in Xcode and on the machine
4. Run `flutter pub get`
5. Run `flutter build ios`

### Expected behavior

The app should build without issues.

---

## Additional context

After a long investigation, the build seems to hand after the `Thin Binary` Xcode build phase is run. More specifically it hangs when executing `/bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" embed`

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
2021-02-23T16:06:32.0033130Z ##[group]Run flutter doctor
2021-02-23T16:06:32.0033810Z [36;1mflutter doctor[0m
2021-02-23T16:06:32.0240420Z shell: /bin/bash -e {0}
2021-02-23T16:06:32.0240870Z env:
2021-02-23T16:06:32.0241360Z   MD_APPLE_SDK_ROOT: /Applications/Xcode_12.4.app
2021-02-23T16:06:32.0242090Z   FLUTTER_HOME: /Users/runner/hostedtoolcache/flutter/1.22.6-stable/x64
2021-02-23T16:06:32.0243090Z ##[endgroup]
2021-02-23T16:06:32.2434310Z Downloading Dart SDK from Flutter engine 2f0af3715217a0c2ada72c717d4ed9178d68f6ed...
2021-02-23T16:06:33.3451860Z   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2021-02-23T16:06:33.3453850Z                                  Dload  Upload   Total   Spent    Left  Speed
2021-02-23T16:06:33.3459350Z 
2021-02-23T16:06:33.3461960Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2021-02-23T16:06:33.8998360Z   0  172M    0  877k    0     0  2904k      0  0:01:00 --:--:--  0:01:00 2895k
2021-02-23T16:06:34.8994220Z  28  172M   28 49.4M    0     0  37.9M      0  0:00:04  0:00:01  0:00:03 37.9M
2021-02-23T16:06:35.0558310Z  90  172M   90  156M    0     0  67.8M      0  0:00:02  0:00:02 --:--:-- 67.7M
2021-02-23T16:06:35.0607330Z 100  172M  100  172M    0     0  70.1M      0  0:00:02  0:00:02 --:--:-- 70.0M
2021-02-23T16:06:41.8839490Z Building flutter tool...
2021-02-23T16:06:42.7413600Z Resolving dependencies...
2021-02-23T16:06:45.5530510Z   _fe_analyzer_shared 7.0.0 (16.0.0 available)
2021-02-23T16:06:45.5540250Z   analyzer 0.39.17 (1.0.0 available)
2021-02-23T16:06:45.5543250Z   archive 2.0.13 (3.0.0 available)
2021-02-23T16:06:45.5553780Z   args 1.6.0 (2.0.0 available)
2021-02-23T16:06:45.5555580Z   async 2.5.0-nullsafety.1 (2.5.0 available)
2021-02-23T16:06:45.5557080Z   boolean_selector 2.1.0-nullsafety.1 (2.1.0 available)
2021-02-23T16:06:45.5558040Z   browser_launcher 0.1.7 (0.1.8 available)
2021-02-23T16:06:45.5558870Z   build_daemon 2.1.4 (2.1.8 available)
2021-02-23T16:06:45.5559740Z   built_collection 4.3.2 (5.0.0 available)
2021-02-23T16:06:45.5560540Z   built_value 7.1.0 (8.0.0 available)
2021-02-23T16:06:45.5561790Z   charcode 1.2.0-nullsafety.1 (1.2.0 available)
2021-02-23T16:06:45.5562750Z   checked_yaml 1.0.2 (1.0.4 available)
2021-02-23T16:06:45.5563540Z   cli_util 0.2.0 (0.3.0 available)
2021-02-23T16:06:45.5564690Z   clock 1.1.0-nullsafety.1 (1.1.0 available)
2021-02-23T16:06:45.5566060Z   collection 1.15.0-nullsafety.3 (1.15.0 available)
2021-02-23T16:06:45.5566990Z   completion 0.2.2 (0.2.3 available)
2021-02-23T16:06:45.5567750Z   convert 2.1.1 (3.0.0 available)
2021-02-23T16:06:45.5568590Z   coverage 0.14.1 (0.15.2 available)
2021-02-23T16:06:45.5569360Z   crypto 2.1.5 (3.0.0 available)
2021-02-23T16:06:45.5570080Z   csslib 0.16.2 (0.17.0 available)
2021-02-23T16:06:45.5570880Z   dds 1.4.0 (1.7.5 available)
2021-02-23T16:06:45.5571690Z   devtools 0.8.0+3 (0.9.7+3 available)
2021-02-23T16:06:45.5572460Z   devtools_server 0.8.0+3 (0.9.7+3 available)
2021-02-23T16:06:45.5573710Z   devtools_shared 0.8.0+3 (0.9.7+3 available)
2021-02-23T16:06:45.5574520Z   dwds 6.0.0 (8.0.3 available)
2021-02-23T16:06:45.5575710Z   fake_async 1.2.0-nullsafety.1 (1.2.0 available)
2021-02-23T16:06:45.5577080Z   file 6.0.0-nullsafety.2 (6.1.0 available)
2021-02-23T16:06:45.5577970Z   file_testing 2.1.0 (3.0.0 available)
2021-02-23T16:06:45.5578710Z   fixnum 0.10.11 (1.0.0 available)
2021-02-23T16:06:45.5579560Z   flutter_template_images 1.0.1
2021-02-23T16:06:45.5580330Z   glob 1.2.0 (2.0.0 available)
2021-02-23T16:06:45.5581060Z   html 0.14.0+3 (0.15.0 available)
2021-02-23T16:06:45.5586730Z   http 0.12.2 (0.13.0 available)
2021-02-23T16:06:45.5587520Z   http_multi_server 2.2.0
2021-02-23T16:06:45.5588300Z   http_parser 3.1.4 (4.0.0 available)
2021-02-23T16:06:45.5589090Z   intl 0.16.1 (0.17.0 available)
2021-02-23T16:06:45.5589800Z   io 0.3.4
2021-02-23T16:06:45.5591040Z   js 0.6.3-nullsafety.1 (0.6.3 available)
2021-02-23T16:06:45.5591960Z   json_annotation 3.1.0 (4.0.0 available)
2021-02-23T16:06:45.5592710Z   json_rpc_2 2.2.2
2021-02-23T16:06:45.5601300Z   logging 0.11.4 (1.0.0 available)
2021-02-23T16:06:45.5645890Z   matcher 0.12.10-nullsafety.1 (0.12.10 available)
2021-02-23T16:06:45.5660100Z   meta 1.3.0-nullsafety.3 (1.3.0 available)
2021-02-23T16:06:45.5675230Z   mime 0.9.7 (1.0.0 available)
2021-02-23T16:06:45.6110330Z   mockito 4.1.1 (4.1.4 available)
2021-02-23T16:06:45.6411320Z   multicast_dns 0.2.2
2021-02-23T16:06:45.6512910Z   mustache_template 1.0.0+1
2021-02-23T16:06:45.6613500Z   native_stack_traces 0.3.7 (0.4.0 available)
2021-02-23T16:06:45.6714990Z   node_interop 1.1.1 (1.2.1 available)
2021-02-23T16:06:45.6815510Z   node_io 1.1.1 (1.2.0 available)
2021-02-23T16:06:45.6916790Z   node_preamble 1.4.12 (1.4.13 available)
2021-02-23T16:06:45.7017380Z   package_config 1.9.3 (2.0.0 available)
2021-02-23T16:06:45.7120710Z   path 1.8.0-nullsafety.1 (1.8.0 available)
2021-02-23T16:06:45.7223960Z   pedantic 1.10.0-nullsafety.1 (1.10.0 available)
2021-02-23T16:06:45.7326010Z   petitparser 3.1.0 (4.0.2 available)
2021-02-23T16:06:45.7490150Z   platform 3.0.0-nullsafety.2 (3.0.0 available)
2021-02-23T16:06:45.7592620Z   pool 1.5.0-nullsafety.1 (1.5.0 available)
2021-02-23T16:06:45.7695740Z   process 4.0.0-nullsafety.2 (4.1.0 available)
2021-02-23T16:06:45.7870280Z   pub_semver 1.4.4 (2.0.0 available)
2021-02-23T16:06:45.7972130Z   pubspec_parse 0.1.5 (0.1.8 available)
2021-02-23T16:06:45.8074480Z   quiver 2.1.3 (3.0.0 available)
2021-02-23T16:06:45.8174950Z   shelf 0.7.5 (1.0.0 available)
2021-02-23T16:06:45.8474420Z   shelf_packages_handler 2.0.0 (2.0.1 available)
2021-02-23T16:06:45.8575860Z   shelf_proxy 0.1.0+7
2021-02-23T16:06:45.8676220Z   shelf_static 0.2.8 (0.2.9+2 available)
2021-02-23T16:06:45.8786100Z   shelf_web_socket 0.2.3 (0.2.4+1 available)
2021-02-23T16:06:45.8843750Z   source_map_stack_trace 2.1.0-nullsafety.2 (2.1.0 available)
2021-02-23T16:06:45.8931950Z   source_maps 0.10.10-nullsafety.1 (0.10.10 available)
2021-02-23T16:06:45.9032730Z   source_span 1.8.0-nullsafety.2 (1.8.1 available)
2021-02-23T16:06:45.9089240Z   sse 3.5.0 (3.8.1 available)
2021-02-23T16:06:45.9093650Z   stack_trace 1.10.0-nullsafety.1 (1.10.0 available)
2021-02-23T16:06:45.9116500Z   stream_channel 2.1.0-nullsafety.1 (2.1.0 available)
2021-02-23T16:06:45.9217920Z   stream_transform 1.2.0 (2.0.0 available)
2021-02-23T16:06:45.9319280Z   string_scanner 1.1.0-nullsafety.1 (1.1.0 available)
2021-02-23T16:06:45.9357540Z   sync_http 0.2.0 (0.3.0 available)
2021-02-23T16:06:45.9415350Z   term_glyph 1.2.0-nullsafety.1 (1.2.0 available)
2021-02-23T16:06:45.9474550Z   test 1.16.0-nullsafety.5 (1.16.3 available)
2021-02-23T16:06:45.9519310Z   test_api 0.2.19-nullsafety.2 (0.2.19 available)
2021-02-23T16:06:45.9548010Z   test_core 0.3.12-nullsafety.5 (0.3.13 available)
2021-02-23T16:06:45.9574620Z   typed_data 1.3.0-nullsafety.3 (1.3.0 available)
2021-02-23T16:06:45.9582240Z   usage 3.4.2 (4.0.0 available)
2021-02-23T16:06:45.9583990Z   uuid 2.2.2 (3.0.0 available)
2021-02-23T16:06:45.9585260Z   vm_service 4.2.0 (6.1.0+1 available)
2021-02-23T16:06:45.9587280Z   vm_snapshot_analysis 0.5.5 (0.5.6 available)
2021-02-23T16:06:45.9589230Z   watcher 0.9.7+15 (1.0.0 available)
2021-02-23T16:06:45.9590880Z   web_socket_channel 1.1.0 (2.0.0 available)
2021-02-23T16:06:45.9592760Z   webdriver 2.1.2 (3.0.0 available)
2021-02-23T16:06:45.9600100Z   webkit_inspection_protocol 0.7.3 (0.7.5 available)
2021-02-23T16:06:45.9603200Z   xml 4.5.1 (5.0.2 available)
2021-02-23T16:06:45.9604720Z   yaml 2.2.1 (3.0.0 available)
2021-02-23T16:06:45.9605780Z Downloading node_preamble 1.4.12...
2021-02-23T16:06:45.9606450Z Downloading json_annotation 3.1.0...
2021-02-23T16:06:45.9606990Z Downloading checked_yaml 1.0.2...
2021-02-23T16:06:45.9607480Z Downloading pubspec_parse 0.1.5...
2021-02-23T16:06:45.9608850Z Downloading test 1.16.0-nullsafety.5...
2021-02-23T16:06:45.9611840Z Downloading file_testing 2.1.0...
2021-02-23T16:06:45.9613060Z Downloading mockito 4.1.1...
2021-02-23T16:06:45.9614950Z Downloading collection 1.15.0-nullsafety.3...
2021-02-23T16:06:45.9616790Z Downloading web_socket_channel 1.1.0...
2021-02-23T16:06:45.9621370Z Downloading watcher 0.9.7+15...
2021-02-23T16:06:45.9622760Z Downloading uuid 2.2.2...
2021-02-23T16:06:45.9624900Z Downloading typed_data 1.3.0-nullsafety.3...
2021-02-23T16:06:45.9626480Z Downloading term_glyph 1.2.0-nullsafety.1...
2021-02-23T16:06:45.9628300Z Downloading sync_http 0.2.0...
2021-02-23T16:06:45.9629990Z Downloading string_scanner 1.1.0-nullsafety.1...
2021-02-23T16:06:45.9631180Z Downloading stream_transform 1.2.0...
2021-02-23T16:06:45.9633800Z Downloading stream_channel 2.1.0-nullsafety.1...
2021-02-23T16:06:45.9634470Z Downloading sse 3.5.0...
2021-02-23T16:06:45.9636180Z Downloading source_span 1.8.0-nullsafety.2...
2021-02-23T16:06:45.9638420Z Downloading source_maps 0.10.10-nullsafety.1...
2021-02-23T16:06:45.9641190Z Downloading source_map_stack_trace 2.1.0-nullsafety.2...
2021-02-23T16:06:45.9642460Z Downloading shelf_web_socket 0.2.3...
2021-02-23T16:06:45.9644330Z Downloading shelf_static 0.2.8...
2021-02-23T16:06:45.9646650Z Downloading shelf_proxy 0.1.0+7...
2021-02-23T16:06:45.9648120Z Downloading shelf_packages_handler 2.0.0...
2021-02-23T16:06:45.9649570Z Downloading quiver 2.1.3...
2021-02-23T16:06:45.9650760Z Downloading pub_semver 1.4.4...
2021-02-23T16:06:45.9652960Z Downloading pool 1.5.0-nullsafety.1...
2021-02-23T16:06:45.9654890Z Downloading platform 3.0.0-nullsafety.2...
2021-02-23T16:06:45.9656080Z Downloading petitparser 3.1.0...
2021-02-23T16:06:45.9658290Z Downloading pedantic 1.10.0-nullsafety.1...
2021-02-23T16:06:45.9660980Z Downloading path 1.8.0-nullsafety.1...
2021-02-23T16:06:45.9662030Z Downloading node_io 1.1.1...
2021-02-23T16:06:45.9663880Z Downloading node_interop 1.1.1...
2021-02-23T16:06:45.9664880Z Downloading mime 0.9.7...
2021-02-23T16:06:45.9666920Z Downloading matcher 0.12.10-nullsafety.1...
2021-02-23T16:06:45.9669190Z Downloading logging 0.11.4...
2021-02-23T16:06:45.9670500Z Downloading json_rpc_2 2.2.2...
2021-02-23T16:06:45.9672130Z Downloading js 0.6.3-nullsafety.1...
2021-02-23T16:06:45.9673300Z Downloading io 0.3.4...
2021-02-23T16:06:45.9674890Z Downloading http_parser 3.1.4...
2021-02-23T16:06:45.9676070Z Downloading http_multi_server 2.2.0...
2021-02-23T16:06:45.9677650Z Downloading html 0.14.0+3...
2021-02-23T16:06:45.9678650Z Downloading glob 1.2.0...
2021-02-23T16:06:45.9680360Z Downloading fixnum 0.10.11...
2021-02-23T16:06:45.9681850Z Downloading devtools_shared 0.8.0+3...
2021-02-23T16:06:45.9683930Z Downloading devtools_server 0.8.0+3...
2021-02-23T16:06:45.9685120Z Downloading devtools 0.8.0+3...
2021-02-23T16:06:45.9686600Z Downloading csslib 0.16.2...
2021-02-23T16:06:45.9689160Z Downloading convert 2.1.1...
2021-02-23T16:06:45.9690530Z Downloading clock 1.1.0-nullsafety.1...
2021-02-23T16:06:45.9693780Z Downloading cli_util 0.2.0...
2021-02-23T16:06:45.9695420Z Downloading charcode 1.2.0-nullsafety.1...
2021-02-23T16:06:45.9697210Z Downloading built_value 7.1.0...
2021-02-23T16:06:45.9698430Z Downloading built_collection 4.3.2...
2021-02-23T16:06:45.9699730Z Downloading browser_launcher 0.1.7...
2021-02-23T16:06:45.9702050Z Downloading boolean_selector 2.1.0-nullsafety.1...
2021-02-23T16:06:45.9707540Z Downloading async 2.5.0-nullsafety.1...
2021-02-23T16:06:45.9708870Z Downloading analyzer 0.39.17...
2021-02-23T16:06:45.9711620Z Downloading _fe_analyzer_shared 7.0.0...
2021-02-23T16:06:45.9713190Z Downloading vm_service 4.2.0...
2021-02-23T16:06:45.9714880Z Downloading test_core 0.3.12-nullsafety.5...
2021-02-23T16:06:45.9716230Z Downloading test_api 0.2.19-nullsafety.2...
2021-02-23T16:06:45.9717260Z Downloading vm_snapshot_analysis 0.5.5...
2021-02-23T16:06:45.9718110Z Downloading shelf 0.7.5...
2021-02-23T16:06:45.9719250Z Downloading build_daemon 2.1.4...
2021-02-23T16:06:45.9720930Z Downloading native_stack_traces 0.3.7...
2021-02-23T16:06:45.9721850Z Downloading yaml 2.2.1...
2021-02-23T16:06:45.9723050Z Downloading xml 4.5.1...
2021-02-23T16:06:45.9726060Z Downloading webkit_inspection_protocol 0.7.3...
2021-02-23T16:06:45.9727020Z Downloading webdriver 2.1.2...
2021-02-23T16:06:45.9727850Z Downloading usage 3.4.2...
2021-02-23T16:06:45.9730050Z Downloading stack_trace 1.10.0-nullsafety.1...
2021-02-23T16:06:45.9732370Z Downloading fake_async 1.2.0-nullsafety.1...
2021-02-23T16:06:45.9733920Z Downloading process 4.0.0-nullsafety.2...
2021-02-23T16:06:45.9735270Z Downloading package_config 1.9.3...
2021-02-23T16:06:45.9736600Z Downloading mustache_template 1.0.0+1...
2021-02-23T16:06:45.9737870Z Downloading multicast_dns 0.2.2...
2021-02-23T16:06:45.9739860Z Downloading meta 1.3.0-nullsafety.3...
2021-02-23T16:06:45.9743580Z Downloading intl 0.16.1...
2021-02-23T16:06:45.9744790Z Downloading http 0.12.2...
2021-02-23T16:06:45.9758310Z Downloading flutter_template_images 1.0.1...
2021-02-23T16:06:45.9763530Z Downloading file 6.0.0-nullsafety.2...
2021-02-23T16:06:45.9765100Z Downloading crypto 2.1.5...
2021-02-23T16:06:45.9767630Z Downloading coverage 0.14.1...
2021-02-23T16:06:45.9768580Z Downloading completion 0.2.2...
2021-02-23T16:06:45.9769920Z Downloading dwds 6.0.0...
2021-02-23T16:06:45.9770500Z Downloading dds 1.4.0...
2021-02-23T16:06:45.9771770Z Downloading args 1.6.0...
2021-02-23T16:06:45.9772940Z Downloading archive 2.0.13...
2021-02-23T16:06:52.6946030Z No dependencies changed.
2021-02-23T16:06:52.7047140Z 83 packages have newer versions incompatible with dependency constraints.
2021-02-23T16:06:52.7148910Z Try `pub outdated` for more information.
2021-02-23T16:07:25.0844380Z 
2021-02-23T16:07:25.0909370Z   ╔════════════════════════════════════════════════════════════════════════════╗
2021-02-23T16:07:25.0911320Z   ║                 Welcome to Flutter! - https://flutter.dev                  ║
2021-02-23T16:07:25.0912870Z   ║                                                                            ║
2021-02-23T16:07:25.0914810Z   ║ The Flutter tool uses Google Analytics to anonymously report feature usage ║
2021-02-23T16:07:25.0916530Z   ║ statistics and basic crash reports. This data is used to help improve      ║
2021-02-23T16:07:25.0917940Z   ║ Flutter tools over time.                                                   ║
2021-02-23T16:07:25.0919110Z   ║                                                                            ║
2021-02-23T16:07:25.0920440Z   ║ Flutter tool analytics are not sent on the very first run. To disable      ║
2021-02-23T16:07:25.0921970Z   ║ reporting, type 'flutter config --no-analytics'. To display the current    ║
2021-02-23T16:07:25.0923490Z   ║ setting, type 'flutter config'. If you opt out of analytics, an opt-out    ║
2021-02-23T16:07:25.0924930Z   ║ event will be sent, and then no further information will be sent by the    ║
2021-02-23T16:07:25.0926310Z   ║ Flutter tool.                                                              ║
2021-02-23T16:07:25.0927480Z   ║                                                                            ║
2021-02-23T16:07:25.0928790Z   ║ By downloading the Flutter SDK, you agree to the Google Terms of Service.  ║
2021-02-23T16:07:25.0930330Z   ║ Note: The Google Privacy Policy describes how data is handled in this      ║
2021-02-23T16:07:25.0931700Z   ║ service.                                                                   ║
2021-02-23T16:07:25.0932980Z   ║                                                                            ║
2021-02-23T16:07:25.0934280Z   ║ Moreover, Flutter includes the Dart SDK, which may send usage metrics and  ║
2021-02-23T16:07:25.0935710Z   ║ crash reports to Google.                                                   ║
2021-02-23T16:07:25.0936910Z   ║                                                                            ║
2021-02-23T16:07:25.0938100Z   ║ Read about data we send with crash reports:                                ║
2021-02-23T16:07:25.0939620Z   ║ https://flutter.dev/docs/reference/crash-reporting                         ║
2021-02-23T16:07:25.0941020Z   ║                                                                            ║
2021-02-23T16:07:25.0942240Z   ║ See Google's privacy policy:                                               ║
2021-02-23T16:07:25.0944150Z   ║ https://policies.google.com/privacy                                        ║
2021-02-23T16:07:25.0946730Z   ╚════════════════════════════════════════════════════════════════════════════╝
2021-02-23T16:07:25.0947330Z 
2021-02-23T16:07:25.9559710Z Downloading Material fonts...                                       0.8s
2021-02-23T16:07:26.0388330Z Downloading Gradle Wrapper...                                       0.0s
2021-02-23T16:07:26.2632880Z Downloading package sky_engine...                                   0.2s
2021-02-23T16:07:27.2366770Z Downloading flutter_patched_sdk tools...                            0.8s
2021-02-23T16:07:28.3302840Z Downloading flutter_patched_sdk_product tools...                    0.6s
2021-02-23T16:07:30.5197640Z Downloading darwin-x64 tools...                                     1.8s
2021-02-23T16:07:31.3988650Z Downloading libimobiledevice...                                     0.0s
2021-02-23T16:07:31.4377730Z Downloading usbmuxd...                                              0.0s
2021-02-23T16:07:31.4807770Z Downloading libplist...                                             0.0s
2021-02-23T16:07:31.6273650Z Downloading openssl...                                              0.1s
2021-02-23T16:07:31.6902530Z Downloading ios-deploy...                                           0.0s
2021-02-23T16:07:31.8392190Z Downloading darwin-x64/font-subset tools...                         0.1s
2021-02-23T16:07:32.0856170Z Downloading android-arm-profile/darwin-x64 tools...                 0.1s
2021-02-23T16:07:32.3259920Z Downloading android-arm-release/darwin-x64 tools...                 0.1s
2021-02-23T16:07:32.7404340Z Downloading android-arm64-profile/darwin-x64 tools...               0.2s
2021-02-23T16:07:32.9937020Z Downloading android-arm64-release/darwin-x64 tools...               0.1s
2021-02-23T16:07:33.2650080Z Downloading android-x64-profile/darwin-x64 tools...                 0.2s
2021-02-23T16:07:33.5811690Z Downloading android-x64-release/darwin-x64 tools...                 0.2s
2021-02-23T16:07:33.7265810Z Doctor summary (to see all details, run flutter doctor -v):
2021-02-23T16:07:37.2803970Z [✓] Flutter (Channel stable, 1.22.6, on Mac OS X 10.15.7 19H512 darwin-x64, locale en)
2021-02-23T16:07:46.0102960Z [!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
2021-02-23T16:07:46.0104750Z     ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
2021-02-23T16:07:46.0106010Z [✓] Xcode - develop for iOS and macOS (Xcode 12.4)
2021-02-23T16:07:46.0106930Z [!] Android Studio (not installed)
2021-02-23T16:07:46.0107610Z [!] Connected device
2021-02-23T16:07:46.0108270Z     ! No devices available
2021-02-23T16:07:46.0108760Z 
2021-02-23T16:07:46.0109710Z ! Doctor found issues in 3 categories.
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
2021-02-23T17:07:39.4484530Z ##[group]Run flutter pub deps -- --style=compact
2021-02-23T17:07:39.4498710Z [36;1mflutter pub deps -- --style=compact[0m
2021-02-23T17:07:39.4661350Z shell: /bin/bash -e {0}
2021-02-23T17:07:39.4661750Z env:
2021-02-23T17:07:39.4662280Z   MD_APPLE_SDK_ROOT: /Applications/Xcode_12.4.app
2021-02-23T17:07:39.4663000Z   FLUTTER_HOME: /Users/runner/hostedtoolcache/flutter/1.22.6-stable/x64
2021-02-23T17:07:39.4663590Z ##[endgroup]
2021-02-23T17:07:40.5603350Z Dart SDK 2.10.5
2021-02-23T17:07:40.5604350Z Flutter SDK 1.22.6
2021-02-23T17:07:40.5605050Z flutter_ios_build_issue 1.0.0+1
2021-02-23T17:07:40.5605630Z 
2021-02-23T17:07:40.5606380Z dependencies:
2021-02-23T17:07:40.5607960Z - cupertino_icons 1.0.0
2021-02-23T17:07:40.5609890Z - firebase_messaging 7.0.3 [meta flutter firebase_core]
2021-02-23T17:07:40.5611640Z - flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
2021-02-23T17:07:40.5612470Z 
2021-02-23T17:07:40.5613390Z dev dependencies:
2021-02-23T17:07:40.5615310Z - flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]
2021-02-23T17:07:40.5616690Z 
2021-02-23T17:07:40.5617480Z transitive dependencies:
2021-02-23T17:07:40.5618750Z - async 2.5.0-nullsafety.1 [collection]
2021-02-23T17:07:40.5620150Z - boolean_selector 2.1.0-nullsafety.1 [source_span string_scanner]
2021-02-23T17:07:40.5621550Z - characters 1.1.0-nullsafety.3
2021-02-23T17:07:40.5622810Z - charcode 1.2.0-nullsafety.1
2021-02-23T17:07:40.5623980Z - clock 1.1.0-nullsafety.1
2021-02-23T17:07:40.5625240Z - collection 1.15.0-nullsafety.3
2021-02-23T17:07:40.5626600Z - fake_async 1.2.0-nullsafety.1 [clock collection]
2021-02-23T17:07:40.5628120Z - firebase_core 0.5.3 [firebase_core_platform_interface flutter quiver meta firebase_core_web]
2021-02-23T17:07:40.5629860Z - firebase_core_platform_interface 2.1.0 [flutter meta plugin_platform_interface quiver]
2021-02-23T17:07:40.5631810Z - firebase_core_web 0.2.1+1 [firebase_core_platform_interface flutter flutter_web_plugins meta js]
2021-02-23T17:07:40.5633540Z - flutter_web_plugins 0.0.0 [flutter characters collection meta typed_data vector_math]
2021-02-23T17:07:40.5634960Z - js 0.6.2
2021-02-23T17:07:40.5636200Z - matcher 0.12.10-nullsafety.1 [stack_trace]
2021-02-23T17:07:40.5637170Z - meta 1.3.0-nullsafety.3
2021-02-23T17:07:40.5638330Z - path 1.8.0-nullsafety.1
2021-02-23T17:07:40.5639590Z - plugin_platform_interface 1.0.3 [meta]
2021-02-23T17:07:40.5641070Z - quiver 2.1.5 [matcher meta]
2021-02-23T17:07:40.5642220Z - sky_engine 0.0.99
2021-02-23T17:07:40.5643760Z - source_span 1.8.0-nullsafety.2 [charcode collection path term_glyph]
2021-02-23T17:07:40.5645190Z - stack_trace 1.10.0-nullsafety.1 [path]
2021-02-23T17:07:40.5646450Z - stream_channel 2.1.0-nullsafety.1 [async]
2021-02-23T17:07:40.5647890Z - string_scanner 1.1.0-nullsafety.1 [charcode source_span]
2021-02-23T17:07:40.5649240Z - term_glyph 1.2.0-nullsafety.1
2021-02-23T17:07:40.5650910Z - test_api 0.2.19-nullsafety.2 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
2021-02-23T17:07:40.5652620Z - typed_data 1.3.0-nullsafety.3 [collection]
2021-02-23T17:07:40.5653940Z - vector_math 2.1.0-nullsafety.3
```

</details>

---

**Comment by markusaksli-nc** [comment]
Hi @daniil-shumko 
Could you please try upgrading to the latest non-nullsafety version `firebase_messaging: ^8.0.0-dev.14`?
https://firebase.flutter.dev/docs/migration#messaging
Thank you

**Comment by daniil-shumko** [comment]
@markusaksli-nc I have tried that but it still did not work. But after many hours I have found what the issue is.
The keychain is getting locked automatically and when `flutter build ios` command tries to run the `xcodebuild` command, it cannot access the keychain to sign the app and this hangs/freezes because it is waiting for the keychain to be unlocked.

A workaround that I have found was to run the `flutter build ios` command with `--no-codesign` which will build the app successfully without the code signing. And then unlock the keychain and run my own `xcodebuild` command to build and sign the app correctly and it seems to work without issues. 

I still do not understand why without the `firebase_messaging` package the app was building fine without the workaround.
